tests/src/main/java/net/neoforged/neoforge/debug/client/TextureAtlasTests.java

```diff
@@ -60,4 +75,64 @@ static void testMaterialAtlas(final DynamicTest test) {
             });
         });
     }
+
+    @TestHolder(description = { "Tests that custom sprite metadata sections get passed through resource reloading properly" }, enabledByDefault = true)
+    @GameTest
+    @EmptyTemplate
+    static void defaultSpriteMetadataSections(final DynamicTest test) {
+        String modId = test.createModId();
+
+        var testResource = ResourceLocation.fromNamespaceAndPath(modId, "resource.png");
+        var sectionType = new MetadataSectionType<>("default_metadata_test", Codec.BOOL);
+
+        var provider = new SpriteSource() {
+            private final MapCodec<? extends SpriteSource> CODEC = MapCodec.unit(this);
+
+            @Override
+            public void run(final ResourceManager manager, final Output output) {
+                var resource = manager.getResource(testResource)
+                        .orElseThrow();
+                output.add(testResource, resource);
+            }
+
+            @Override
+            public MapCodec<? extends SpriteSource> codec() {
+                return CODEC;
+            }
+        };
+
+        test.framework().modEventBus().addListener(AddPackFindersEvent.class, event -> {
+            event.addPackFinders(
+                    ResourceLocation.fromNamespaceAndPath("neotests", "assets/" + modId + "/test_pack"),
+                    PackType.CLIENT_RESOURCES,
+                    Component.literal("Sprite metadata test pack"),
+                    PackSource.BUILT_IN,
+                    true,
+                    Position.TOP);
+        });
+
+        test.framework().modEventBus().addListener(RegisterSpriteSourcesEvent.class, event -> {
+            event.register(ResourceLocation.fromNamespaceAndPath(modId, "test_sprite"), provider.codec());
+        });
+
+        test.framework().modEventBus().addListener(RegisterMetadataSectionTypesEvent.class, event -> {
+            event.register(sectionType);
+        });
+
+        test.onGameTest(helper -> {
```

Comment:
The game test for a client test is weird and will probably never run. Maybe look at the test above which seems to test a similar thing?

Reply:
Yeah, I can probably look for the resource reload finishing - I wrote it this way just so I could confirm that the resource packs were enabled/loaded for my sanity :sweat_smile: